### DISCUSSION TITLE
Startup Warnings in log

### DISCUSSION THREAD

**Discussion by RBEmerson970** [remsky/Kokoro-FastAPI#158] [Q&A] [answered]
Do these warnings (copied from full log) matter?

```
2025-02-11 13:06:31 /app/.venv/lib/python3.10/site-packages/torch/nn/modules/rnn.py:123: UserWarning: dropout option adds dropout after all but last recurrent layer, so non-zero dropout expects num_layers greater than 1, but got dropout=0.2 and num_layers=1
2025-02-11 13:06:31   warnings.warn(
2025-02-11 13:06:31 /app/.venv/lib/python3.10/site-packages/torch/nn/utils/weight_norm.py:143: FutureWarning: `torch.nn.utils.weight_norm` is deprecated in favor of `torch.nn.utils.parametrizations.weight_norm`.
2025-02-11 13:06:31   WeightNorm.apply(module, name, dim)
```

```
2025-02-11 13:06:21 
2025-02-11 13:06:21 ==========
2025-02-11 13:06:21 == CUDA ==
2025-02-11 13:06:21 ==========
2025-02-11 13:06:21 
2025-02-11 13:06:21 CUDA Version 12.4.1
2025-02-11 13:06:21 
2025-02-11 13:06:21 Container image Copyright (c) 2016-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
2025-02-11 13:06:21 
2025-02-11 13:06:21 This container image and its contents are governed by the NVIDIA Deep Learning Container License.
2025-02-11 13:06:21 By pulling and using the container, you accept the terms and conditions of this license:
2025-02-11 13:06:21 https://developer.nvidia.com/ngc/nvidia-deep-learning-container-license
2025-02-11 13:06:21 
2025-02-11 13:06:21 A copy of this license is made available in this container at /NGC-DL-CONTAINER-LICENSE for your convenience.
2025-02-11 13:06:21 
2025-02-11 13:06:31 06:06:31 PM | INFO     | main:57 | Loading TTS model and voice packs...
2025-02-11 13:06:31 06:06:31 PM | INFO     | model_manager:38 | Initializing Kokoro V1 on cuda
2025-02-11 13:06:31 06:06:31 PM | DEBUG    | paths:101 | Searching for model in path: /app/api/src/models
2025-02-11 13:06:22 2025-02-11 18:06:22.012 | INFO     | __main__:download_model:60 - Model files already exist and are valid
2025-02-11 13:06:22    Building kokoro-fastapi @ file:///app
2025-02-11 13:06:22       Built kokoro-fastapi @ file:///app
2025-02-11 13:06:22 Uninstalled 1 package in 0.69ms
2025-02-11 13:06:22 Installed 1 package in 0.45ms
2025-02-11 13:06:31 INFO:     Started server process [67]
2025-02-11 13:06:31 INFO:     Waiting for application startup.
2025-02-11 13:06:31 /app/.venv/lib/python3.10/site-packages/torch/nn/modules/rnn.py:123: UserWarning: dropout option adds dropout after all but last recurrent layer, so non-zero dropout expects num_layers greater than 1, but got dropout=0.2 and num_layers=1
2025-02-11 13:06:31   warnings.warn(
2025-02-11 13:06:31 /app/.venv/lib/python3.10/site-packages/torch/nn/utils/weight_norm.py:143: FutureWarning: `torch.nn.utils.weight_norm` is deprecated in favor of `torch.nn.utils.parametrizations.weight_norm`.
2025-02-11 13:06:31   WeightNorm.apply(module, name, dim)
2025-02-11 13:06:34 INFO:     Application startup complete.
2025-02-11 13:06:34 INFO:     Uvicorn running on http://0.0.0.0:8880 (Press CTRL+C to quit)
2025-02-11 13:10:38 INFO:     Shutting down
2025-02-11 13:10:38 INFO:     Waiting for application shutdown.
2025-02-11 13:10:38 INFO:     Application shutdown complete.
2025-02-11 13:10:38 INFO:     Finished server process [67]
2025-02-11 13:06:31 06:06:31 PM | INFO     | kokoro_v1:45 | Loading Kokoro model on cuda
2025-02-11 13:06:31 06:06:31 PM | INFO     | kokoro_v1:46 | Config path: /app/api/src/models/v1_0/config.json
2025-02-11 13:06:31 06:06:31 PM | INFO     | kokoro_v1:47 | Model path: /app/api/src/models/v1_0/kokoro-v1_0.pth
2025-02-11 13:06:32 06:06:32 PM | DEBUG    | paths:153 | Scanning for voices in path: /app/api/src/voices/v1_0
2025-02-11 13:06:32 06:06:32 PM | DEBUG    | paths:131 | Searching for voice in path: /app/api/src/voices/v1_0
2025-02-11 13:06:32 06:06:32 PM | DEBUG    | model_manager:77 | Using default voice 'af_heart' for warmup
2025-02-11 13:06:32 06:06:32 PM | INFO     | kokoro_v1:73 | Creating new pipeline for language code: a
2025-02-11 13:06:33 06:06:33 PM | DEBUG    | kokoro_v1:239 | Generating audio for text with lang_code 'a': 'Warmup text for initialization....'
2025-02-11 13:06:34 06:06:34 PM | DEBUG    | kokoro_v1:246 | Got audio chunk with shape: torch.Size([57600])
2025-02-11 13:06:34 06:06:34 PM | INFO     | model_manager:84 | Warmup completed in 3832ms
2025-02-11 13:06:34 06:06:34 PM | INFO     | main:101 | 
2025-02-11 13:06:34 
2025-02-11 13:06:34 ░░░░░░░░░░░░░░░░░░░░░░░░
2025-02-11 13:06:34 
2025-02-11 13:06:34     ╔═╗┌─┐┌─┐┌┬┐
2025-02-11 13:06:34     ╠╣ ├─┤└─┐ │ 
2025-02-11 13:06:34     ╚  ┴ ┴└─┘ ┴
2025-02-11 13:06:34     ╦╔═┌─┐┬┌─┌─┐
2025-02-11 13:06:34     ╠╩╗│ │├┴┐│ │
2025-02-11 13:06:34     ╩ ╩└─┘┴ ┴└─┘
2025-02-11 13:06:34 
2025-02-11 13:06:34 ░░░░░░░░░░░░░░░░░░░░░░░░
2025-02-11 13:06:34                 
2025-02-11 13:06:34 Model warmed up on cuda: kokoro_v1CUDA: True
2025-02-11 13:06:34 67 voice packs loaded
2025-02-11 13:06:34 
2025-02-11 13:06:34 Beta Web Player: http://0.0.0.0:8880/web/
2025-02-11 13:06:34 or http://localhost:8880/web/
2025-02-11 13:06:34 ░░░░░░░░░░░░░░░░░░░░░░░░
2025-02-11 13:06:34 
```

**Comment by remsky** [comment, marked as answer]
Not really a concern, no. 

The dropout warning is indicating that the dropout parameter (0.2) won't have any effect since there's only one layer. It's more of an optimization opportunity than a problem. The weight_norm deprecation warning won't impact anything either, it's just letting us know it'll need to be updated in an upcoming version.

**Reply by RBEmerson970** [reply]
Understood. :)
